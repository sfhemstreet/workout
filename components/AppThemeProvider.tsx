import { ThemeProvider } from "@emotion/react";
import { ReactNode } from "react";

import { darkColors, lightColors } from "../styles/colors";
import { GlobalStyles } from "../styles/GlobalStyles";
import { theme } from "../styles/theme";
import { ThemeType } from "../types/ThemeType";

type AppThemeProviderProps = {
  children: ReactNode;
  themeType: ThemeType;
};
/**
 * AppThemeProvider
 * - provides the current theme in use to child components.
 * - provides the current theme to Emotion styled components via Emotion's ThemeProvider.
 * - applies global CSS via GlobalStyles.
 *
 * @param children
 */
export const AppThemeProvider: React.FC<AppThemeProviderProps> = ({
  children,
  themeType,
}) => (
  <ThemeProvider
    theme={{
      ...theme,
      name: themeType,
      colors: themeType === "LIGHT" ? lightColors : darkColors,
    }}
  >
    <GlobalStyles themeType={themeType} />
    {children}
  </ThemeProvider>
);
