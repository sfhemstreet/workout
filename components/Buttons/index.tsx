export { SubmitButton } from "./SubmitButton";
export { SecondaryButton } from "./SecondaryButton";
export { TertiaryButton } from "./TertiaryButton";
export { Hamburger } from "./Hamburger";
export { SmallRoundButton } from "./SmallRoundButton";
export { ArrowButton } from "./ArrowButton";
export { InlineButton } from "./InlineButton";
export { SelectableInlineButton } from "./SelectableInlineButton";
