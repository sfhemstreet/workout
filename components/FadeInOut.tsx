import { css } from "@emotion/react";
import styled from "@emotion/styled";
import { ReactNode } from "react";
import { Transition, TransitionStatus } from "react-transition-group";

import { FadeInKeyframe } from "../styles/keyframes";
import { transitions } from "../styles/transitions";

type FadeInOutProps = {
  isShowing: boolean;
  alwaysMounted?: boolean;
  children?: ReactNode;
  render?: (transitionStatus: TransitionStatus) => void;
  timeout?:
    | {
        enter?: number;
        exit?: number;
        appear?: number;
      }
    | number;
};

/**
 * FadeInOut wraps a Transition component from `react-transition-group`
 * to fade in and out components with the same timeout and transition settings.
 *
 * @param isShowing controls if component is faded in or not
 * @param alwaysMounted optional, if true keeps component mounted to the DOM while faded out
 * @param children child components to render, they will be wrapped with div that fades
 * @param render optional, render method passes the transitionStatus, overrides children
 * @param timeout optional, times in ms for enter, exit, appear. Default `enter: 300, exit: 800, appear: 300`
 */
export const FadeInOut: React.FC<FadeInOutProps> = ({
  isShowing,
  alwaysMounted = false,
  children,
  render,
  timeout,
}) => {
  // Give default arguments for timeout
  if (!timeout) {
    timeout = { enter: 300, exit: 800, appear: 300 };
  } else if (typeof timeout === "object") {
    // Override defaults
    timeout = { enter: 300, exit: 800, appear: 300, ...timeout };
  }

  return (
    <Transition
      in={isShowing}
      timeout={timeout}
      unmountOnExit={!alwaysMounted}
      mountOnEnter={!alwaysMounted}
      appear={false}
    >
      {(status) =>
        render ? (
          render(status)
        ) : (
          <Container status={status}>{children}</Container>
        )
      }
    </Transition>
  );
};

const Container = styled.div<{ status: TransitionStatus }>`
  width: 100%;
  height: 100%;

  ${(p) => FadeInOutMixin(p.status)};
`;

/**
 * FadeInOutMixin
 * 
 * Adds css properties to fade in and out any HTML element. 
 * 
 * @param transitionStatus 
 */
export const FadeInOutMixin = (transitionStatus: TransitionStatus) => css `
  opacity: ${transitionStatus === "entered"
    ? 1
    : 0};
  animation: ${FadeInKeyframe} 300ms ease-in-out both;
  transition: ${transitions.normal};
`;

/**
 * HTMLDivElement that fades in and out thru the transitionStatus prop
 * using the FadeInOutMixin.
 * 
 * @param transitionStatus
 */
export const FadableDiv = styled.div<{ transitionStatus: TransitionStatus }>`
  ${(p) => FadeInOutMixin(p.transitionStatus)};
`;
