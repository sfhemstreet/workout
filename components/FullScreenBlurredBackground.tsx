import styled from "@emotion/styled";
import React, { useRef } from "react";
import { TransitionStatus } from "react-transition-group";
import { useTrapFocus } from "../hooks/useTrapFocus";

type FullScreenBlurredBackgroundProps = {
  children: React.ReactNode;
  transitionStatus?: TransitionStatus;
  onClick?: () => void;
};

/**
 * FullscreenBlurredBackground
 * 
 * Div that fills entire screen, blurring content below it in the DOM.
 * It also traps focus to make sure that only its children can be 'tabbed' into focus.
 *
 * Note: If browser does not support backdrop-filter the background is just darkened more.
 * 
 * @param children Child elements / components
 * @param transitionStatus controls fade-in/fade-out effect
 * @param onClick optional onClick function
 */
export const FullscreenBlurredBackground: React.FC<FullScreenBlurredBackgroundProps> =
  ({ children, transitionStatus, onClick }) => {
    const ref = useRef<HTMLDivElement>(null);
    useTrapFocus(ref);

    return (
      <FullscreenBlurredBackgroundDiv
        transitionStatus={transitionStatus}
        ref={ref}
        onClick={onClick}
      >
        {children}
      </FullscreenBlurredBackgroundDiv>
    );
  };

/**
 * HTMLDivElement that fills entire screen, blurring content below it in the DOM.
 *
 * If browser does not support backdrop-filter the background is darkened more.
 */
export const FullscreenBlurredBackgroundDiv = styled.div<{
  transitionStatus?: TransitionStatus;
}>`
  position: fixed;
  z-index: 9999;
  top: 0px;
  left: 0px;
  width: 100%;
  height: 100%;
  min-height: 700px;

  overflow-y: scroll;

  background-color: rgba(0, 0, 0, 0.5);

  backdrop-filter: blur(2px);

  display: flex;
  justify-content: center;

  transition: ${(p) => p.theme.transitions.normal};
  opacity: ${(p) =>
    p.transitionStatus ? (p.transitionStatus === "entered" ? 1 : 0) : 1};

  @media ${(p) => p.theme.media.tablet} {
    padding-top: 10px;
  }

  @supports not (backdrop-filter: blur(2px)) {
    background-color: rgba(0, 0, 0, 0.8);
  }
`;
