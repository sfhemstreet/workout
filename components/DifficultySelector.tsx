import styled from "@emotion/styled";
import { SurfaceElevation } from "../styles/SurfaceElevation";
import { Difficulty } from "../types/Difficulty";
import { SelectableInlineButton } from "./Buttons";

const DIFFICULTY_LEVELS: Difficulty[] = ["active-rest", "moderate", "hard", "extreme"];

type DifficultySelectorProps = {
  difficulty: Difficulty;
  onSelect: (difficulty: Difficulty) => void;
};

/**
 * DifficultySelector
 * 
 * Displays all difficulty options as selectable buttons.
 * 
 * @param difficulty the selected difficulty
 * @param onSelect function called when a difficulty is selected
 */
export const DifficultySelector = ({
  difficulty,
  onSelect,
}: DifficultySelectorProps) => (
  <Row>
    {DIFFICULTY_LEVELS.map((diff) => (
      <SelectableInlineButton
        key={diff}
        isSelected={difficulty === diff}
        onClick={(e) => {
          e.preventDefault();
          onSelect(diff);
        }}
      >
        {diff}
      </SelectableInlineButton>
    ))}
  </Row>
);

const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  margin: 10px;
  padding: 10px;

  border-radius: 10px;
  ${(p) => SurfaceElevation(p.theme.name, 3)};
`;
