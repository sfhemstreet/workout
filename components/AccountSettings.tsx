import styled from "@emotion/styled";
import React, { useState } from "react";
import {
  SwitchTransition,
  Transition,
  TransitionStatus,
} from "react-transition-group";
import { SurfaceElevation } from "../styles/SurfaceElevation";
import { User } from "../types";
import { InlineButton, SecondaryButton, SubmitButton } from "./Buttons";
import { FadeInOutMixin } from "./FadeInOut";
import {
  EMPTY_USERNAME,
  USERNAME_LENGTH,
  USERNAME_TAKEN,
} from "./SignIn/constants";
import { checkUsername } from "./SignIn/helpers";
import { TextInput } from "./TextInput";
import { H1, P } from "./Txt";

type AccountSettingsProps = {
  user: User;

  username: {
    onSubmit: (name: string) => void;
    onTypeName: (name: string) => void;
    isUnique: boolean;
    error: string | undefined;
  };

  avatar: {
    onSubmit: (url: string) => void;
    onCheckUrl: (url: string) => void;
    isUrlOk: boolean;
    error: string | undefined;
  };

  onDeleteAccount: () => void;
  onClose: () => void;
};

/**
 * AccountSettings
 * 
 * Allows signed in users to change their username / avatar, 
 * and for all user's to delete their account.
 * 
 * @param user current User
 * @param username object with properties concerning changing of username
 * @param avatar object with properties concerning changing of avatar
 * @param onDeleteAccount function called after user confirms to delete their account
 * @param onClose function called when user wishes to exit settings
 */
export const AccountSettings = ({
  user,
  username,
  avatar,
  onDeleteAccount,
  onClose,
}: AccountSettingsProps) => {
  const [newUsername, setNewUsername] = useState(user.name);
  const [usernameError, setUsernameError] = useState<string | undefined>(
    undefined
  );
  const [newAvatarUrl, setNewAvatarUrl] = useState(user.avatar);
  const [avatarError, setAvatarError] = useState<string | undefined>(undefined);

  const [isDeletingAccount, setIsDeletingAccount] = useState(false);

  const handleSubmitUsername = () => {
    if (newUsername.trim().length === 0) {
      setUsernameError(EMPTY_USERNAME);
      return;
    }

    if (newUsername.trim() === user.name) {
      setUsernameError("You are already using that name...");
      return;
    }

    if (!username.isUnique) {
      setUsernameError(USERNAME_TAKEN);
      return;
    }

    const { hasLength, hasNoSymbols, hasNoWhitespace } = checkUsername(
      newUsername.trim()
    );

    if (!hasLength) {
      setUsernameError(
        `Name must be ${USERNAME_LENGTH.min}-${USERNAME_LENGTH.max} characters.`
      );
      return;
    }
    if (!hasNoSymbols) {
      setUsernameError(`Name cannot have symbols, besides '-' and '_'`);
      return;
    }
    if (!hasNoWhitespace) {
      setUsernameError(`Name cannot contain whitespace.`);
      return;
    }

    username.onSubmit(newUsername.trim());
  };

  const handleNameChange = (name: string) => {
    setNewUsername(name);
    user.name !== name && username.onTypeName(name.trim());

    if (usernameError) setUsernameError(undefined);
  };

  const handleSubmitAvatar = () => {
    if (newAvatarUrl.trim().length === 0) {
      setAvatarError("Please enter URL to image.");
      return;
    }
    if (!avatar.isUrlOk) {
      setAvatarError("URL must be an image.");
      return;
    }
    if (newAvatarUrl.trim() === user.avatar) {
      setAvatarError("You are already using this as your avatar...");
      return;
    }

    avatar.onSubmit(newAvatarUrl.trim());
  };

  const handleAvatarChange = (url: string) => {
    setNewAvatarUrl(url);
    user.avatar !== url && avatar.onCheckUrl(url);

    if (avatarError) setAvatarError(undefined);
  };

  return (
    <SwitchTransition mode="out-in">
      <Transition
        timeout={{ enter: 0, appear: 100, exit: 300 }}
        key={isDeletingAccount ? "delete" : "edit"}
      >
        {(transitionStatus) =>
          isDeletingAccount ? (
            <Container transitionStatus={transitionStatus}>
              <CloseBtn onClick={onClose}>✕</CloseBtn>
              <H1>Delete Account</H1>
              <P>Are you sure you want to delete your account?</P>
              <Row>
                <SecondaryButton onClick={() => setIsDeletingAccount(false)}>
                  Cancel
                </SecondaryButton>
                <SubmitButton onClick={onDeleteAccount}>Delete</SubmitButton>
              </Row>
            </Container>
          ) : (
            <Container transitionStatus={transitionStatus}>
              <CloseBtn onClick={onClose}>✕</CloseBtn>
              <H1>Account Settings</H1>
              {!user.isAuthenticated && (
                <P>
                  You need to be logged in to change your username or avatar.
                </P>
              )}
              <TextInput
                id="account-settings-name"
                label="Username"
                onChange={handleNameChange}
                placeholder="Want a new username?"
                value={newUsername}
                error={
                  !username.isUnique && newUsername !== user.name
                    ? USERNAME_TAKEN
                    : usernameError
                }
                isOutlined
                isDisabled={!user.isAuthenticated}
              />
              <Row flexEnd>
                <SecondaryButton
                  onClick={handleSubmitUsername}
                  disabled={!user.isAuthenticated}
                >
                  Change Username
                </SecondaryButton>
              </Row>

              <TextInput
                id="account-settings-avatar"
                label="Avatar URL"
                onChange={handleAvatarChange}
                placeholder="Image URl"
                value={newAvatarUrl}
                error={
                  !avatar.isUrlOk
                    ? "Invalid URL, must be an image."
                    : avatarError
                }
                isOutlined
                isDisabled={!user.isAuthenticated}
              />
              <Row flexEnd>
                <SecondaryButton
                  onClick={handleSubmitAvatar}
                  disabled={!user.isAuthenticated}
                >
                  Change Avatar
                </SecondaryButton>
              </Row>
              <br />
              <Row>
                <SecondaryButton onClick={onClose}>Close</SecondaryButton>
                <SecondaryButton onClick={() => setIsDeletingAccount(true)}>
                  Delete Account
                </SecondaryButton>
              </Row>
            </Container>
          )
        }
      </Transition>
    </SwitchTransition>
  );
};

const Container = styled.article<{ transitionStatus: TransitionStatus }>`
  position: relative;
  max-width: 300px;
  width: 100%;
  height: fit-content;

  border-radius: 20px;
  ${(p) => SurfaceElevation(p.theme.name, 3)};

  padding: 1rem;

  ${(p) => FadeInOutMixin(p.transitionStatus)};

  @media ${(p) => p.theme.media.tablet} {
    max-width: 500px;
  }

  button {
    margin-top: 0.25rem;
    margin-bottom: 1rem;
  }
`;

const Row = styled.div<{ flexEnd?: boolean }>`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: ${(p) => (p.flexEnd ? "flex-end" : "space-around")};
  align-items: center;
`;

const CloseBtn = styled(InlineButton)`
  position: absolute;
  top: 2px;
  right: 12px;
  font-family: ${(p) => p.theme.font.numberFamily};
  font-size: 2rem;
  background: transparent;
`;
