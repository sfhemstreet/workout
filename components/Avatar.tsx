import styled from "@emotion/styled";
import {
  SwitchTransition,
  Transition,
  TransitionStatus,
} from "react-transition-group";
import { FadeInKeyframe } from "../styles/keyframes";

import { SurfaceElevation } from "../styles/SurfaceElevation";
import { FadeInOutMixin } from "./FadeInOut";
import { LoadingShimmer } from "./LoadingShimmer";


type AvatarProps = {
  isLoading: boolean;
  src: string;
  onClick?: () => void;
};

/**
 * Avatar
 * 
 * Displays round user avatar 
 * 
 * @param isLoading when `true` a LoadingShimmer is rendered instead of avatar
 * @param src src of avatar img
 * @param onClick function to call when avatar is clicked
 */
export const Avatar = ({ isLoading, src, onClick }: AvatarProps) => (
  <BtnContainer
    noHover={onClick === undefined}
    onClick={() => !isLoading && onClick && onClick()}
    aria-label="User information"
  >
    <SwitchTransition mode="out-in">
      <Transition key={isLoading ? "true" : "false"} timeout={200}>
        {(status) => (
          <>
            {isLoading ? (
              <LoadingContainer status={status}>
                <LoadingShimmer />
              </LoadingContainer>
            ) : (
              <Img
                role="button"
                status={status}
                src={src}
                alt="User Avatar"
                title="Toggle user menu"
              />
            )}
          </>
        )}
      </Transition>
    </SwitchTransition>
  </BtnContainer>
);

const BtnContainer = styled.button<{ noHover: boolean }>`
  padding: 0px;
  border: none;
  overflow: hidden;

  ${(p) => SurfaceElevation(p.theme.name, 4)};

  width: 50px;
  height: 50px;
  border-radius: 50%;

  display: flex;
  justify-content: space-around;
  align-items: center;

  animation: ${FadeInKeyframe} 300ms linear both;
  transition: ${(p) => p.theme.transitions.normal};

  cursor: ${(p) => (p.noHover ? "default" : "pointer")};

  :hover {
    ${(p) => !p.noHover && `background: ${p.theme.colors.secondary}`};
  }

  @media ${(p) => p.theme.media.tablet} {
    width: 60px;
    height: 60px;
  }
`;

const Img = styled.img<{ status: TransitionStatus }>`
  width: 46px;
  height: 46px;
  border-radius: 50%;

  ${(p) => FadeInOutMixin(p.status)};

  @media ${(p) => p.theme.media.tablet} {
    width: 56px;
    height: 56px;
  }
`;

const LoadingContainer = styled.div<{ status: TransitionStatus }>`
  width: 46px;
  height: 46px;
  border-radius: 50%;

  overflow: hidden;

  display: flex;
  justify-content: space-around;
  align-items: center;

  transition: ${(p) => p.theme.transitions.normal};
  opacity: ${(p) => (p.status === "entered" ? 1 : 0)};

  @media ${(p) => p.theme.media.tablet} {
    width: 56px;
    height: 56px;
  }
`;
