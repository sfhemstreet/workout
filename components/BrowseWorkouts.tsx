import { css } from "@emotion/react";
import styled from "@emotion/styled";
import React, { useState } from "react";
import { SwitchTransition, Transition } from "react-transition-group";

import {
  useBrowseWorkouts,
  WorkoutsSortType,
} from "../hooks/useBrowseWorkouts";
import { useWorkoutsFilter } from "../hooks/useWorkoutsFilter";
import { ShimmerBackgroundKeyframe } from "../styles/keyframes";
import { SurfaceElevation } from "../styles/SurfaceElevation";
import { FlexSetting } from "../types/FlexSetting";
import { SecondaryButton } from "./Buttons";
import { DifficultySelector } from "./DifficultySelector";
import { Toggler } from "./Toggler";
import { H3, LowEmpSpan } from "./Txt";
import { WorkoutPreview } from "./WorkoutPreview";
import { WorkoutPreviewLoadingShimmers } from "./WorkoutPreviewLoadingShimmers";

export const BrowseWorkouts = React.memo(() => {
  const { filter, onChangeDifficultyFilter } = useWorkoutsFilter();

  const [{ sort, limit }, setOptions] =
    useState<BrowseWorkoutsOptions>(DEFAULT_OPTIONS);

  const { isLoading, workouts, getMoreWorkouts, isExhausted, goToWorkout } =
    useBrowseWorkouts({
      sort,
      limit,
      difficulty: filter.difficulty,
      rating: filter.rating,
    });

  const handleChangeSort = () =>
    setOptions((prevOptions) => ({
      ...prevOptions,
      sort: prevOptions.sort === "LATEST" ? "POPULAR" : "LATEST",
    }));

  const handleChangeLimit = (newLimit: number) =>
    setOptions((prevOptions) => ({ ...prevOptions, limit: newLimit }));

  return (
    <Container>
      <H3>Browse Workouts</H3>
      <Row>
        <Toggler
          onLabel="Latest"
          offLabel="Popular"
          onToggle={handleChangeSort}
          isOn={sort === "LATEST"}
          ariaLabel={`Change sort to ${
            sort === "LATEST" ? "popular" : "latest"
          }`}
          title="Sort"
        />
        <Column>
          <LowEmpSpan padding="1rem 1rem 0rem 1rem">Difficulty</LowEmpSpan>
          <DifficultySelector
            difficulty={filter.difficulty}
            onSelect={onChangeDifficultyFilter}
          />
        </Column>
      </Row>

      <WorkoutsContainer>
        <SwitchTransition mode="out-in">
          <Transition key={isLoading ? "loading" : "results"} timeout={300}>
            {(transitionStatus) =>
              isLoading ? (
                <WorkoutPreviewLoadingShimmers
                  transitionStatus={transitionStatus}
                />
              ) : (
                workouts.map((workout) => (
                  <WorkoutPreview
                    key={`browse-${workout.id}`}
                    workout={workout}
                    onClick={() => goToWorkout(workout.id)}
                  />
                ))
              )
            }
          </Transition>
        </SwitchTransition>
      </WorkoutsContainer>

      {!isExhausted && workouts.length > 0 && (
        <LoadMoreBtn
          disabled={isLoading}
          isLoading={isLoading}
          onClick={getMoreWorkouts}
        >
          More Workouts!
        </LoadMoreBtn>
      )}
    </Container>
  );
});

type BrowseWorkoutsOptions = {
  sort: WorkoutsSortType;
  limit: number;
};

const DEFAULT_OPTIONS: BrowseWorkoutsOptions = {
  sort: "POPULAR",
  limit: 10,
};

const Container = styled.section`
  border-radius: 10px;
  width: 100%;
  height: 100%;
  ${(p) => SurfaceElevation(p.theme.name, 1)};
`;

const WorkoutsContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
`;

const LoadMoreBtn = styled(SecondaryButton)<{
  isLoading: boolean;
}>`
  ${(p) =>
    p.isLoading &&
    css`
      animation: ${ShimmerBackgroundKeyframe} 2s alternate infinite ease-in-out;
      background-image: ${p.theme.colors.superGradient};
      background-size: 400% 100%;
      overflow: hidden;
    `}

  transition: ${(p) => p.theme.transitions.normal};

  margin: 1rem;
`;

export const Row = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  flex-wrap: wrap;
`;

export const Column = styled.div<{
  alignItems?: FlexSetting;
  justifyContent?: FlexSetting;
}>`
  display: flex;
  flex-direction: column;
  justify-content: ${(p) => p.justifyContent ?? "center"};
  align-items: ${(p) => p.alignItems ?? "flex-start"};
`;
