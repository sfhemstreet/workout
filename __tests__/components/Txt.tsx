import * as Txt from "../../components/Txt";
import { render } from "../test-utils/render";

test("Txt components render properly", () => {
  const components = render(
    <div id="txt-test-div">
      <Txt.H1 padding="10000px">Test H1</Txt.H1>
      <Txt.H2 margin="10000rem">Test H2</Txt.H2>
      <Txt.H3>Test H3</Txt.H3>
      <Txt.H4 padding="666px" margin="666px" textAlign="center">Test H4</Txt.H4>
      <Txt.H5 textAlign="right">Test H5</Txt.H5>
      <Txt.LowEmpSpan margin="10000px">Test LowEmpSpan</Txt.LowEmpSpan>
      <Txt.NumberSpan>Test NumberSpan</Txt.NumberSpan>
      <Txt.P>Test P</Txt.P>
    </div>
  );

  expect(components).toMatchSnapshot();
});