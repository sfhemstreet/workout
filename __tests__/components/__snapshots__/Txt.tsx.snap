// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Txt components render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="txt-test-div"
      >
        <h1
          class="css-1kk2lgj-H1-TextMixin e1ez90bh0"
        >
          Test H1
        </h1>
        <h2
          class="css-1hf4ge2-H2-TextMixin ewijl6f0"
        >
          Test H2
        </h2>
        <h3
          class="css-y7h9ws-H3-TextMixin eyl1k3n0"
        >
          Test H3
        </h3>
        <h4
          class="css-4ivc9-H4-TextMixin e1uetepz0"
        >
          Test H4
        </h4>
        <h5
          class="css-1lbf9r5-H5-TextMixin e15034nc0"
        >
          Test H5
        </h5>
        <span
          class="css-wq6nlq-LowEmpSpan-TextMixin ebud7590"
        >
          Test LowEmpSpan
        </span>
        <span
          class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
        >
          Test NumberSpan
        </span>
        <p
          class="css-44y8rz-P-TextMixin elrci8c0"
        >
          Test P
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="txt-test-div"
    >
      <h1
        class="css-1kk2lgj-H1-TextMixin e1ez90bh0"
      >
        Test H1
      </h1>
      <h2
        class="css-1hf4ge2-H2-TextMixin ewijl6f0"
      >
        Test H2
      </h2>
      <h3
        class="css-y7h9ws-H3-TextMixin eyl1k3n0"
      >
        Test H3
      </h3>
      <h4
        class="css-4ivc9-H4-TextMixin e1uetepz0"
      >
        Test H4
      </h4>
      <h5
        class="css-1lbf9r5-H5-TextMixin e15034nc0"
      >
        Test H5
      </h5>
      <span
        class="css-wq6nlq-LowEmpSpan-TextMixin ebud7590"
      >
        Test LowEmpSpan
      </span>
      <span
        class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
      >
        Test NumberSpan
      </span>
      <p
        class="css-44y8rz-P-TextMixin elrci8c0"
      >
        Test P
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
